### DISCUSSION TITLE
Install/Execute Issues

### DISCUSSION THREAD

**Discussion by esnelling** [junzis/pyModeS#95] [Q&A] [answered]
Thanks very much in advance.  I am having difficulty getting 'modeslive' to execute.  Have installed via 'pip install pyModes'.  This is installed on a Raspberry Pi which has successfully been running dump1090-fa for a few years.  I am not entirely unfamiliar with Python but most of my work has been authoring personal Py programs to manipulate data from the aircraft.json off the dump1090-fa server.

pi@raspberrypi:~ $ python --version
Python 2.7.16

Things I have tried so far:

1. Stop dump190-fa
2. Installation by pip3 .  I can't tell from github if this is written for Python 2 or 3.  I assume 2 based on suggestion to use pip vs pip3.
3. Installation of the developmental code
4. Have uninstalled all that and now back to just the pip install pyModeS

Current error on execute is:
pi@raspberrypi:~ $ modeslive --source rtlsdr
Traceback (most recent call last):
  File "/usr/local/bin/modeslive", line 10, in <module>
    from pyModeS.streamer.decode import Decode
  File "/usr/local/lib/python2.7/dist-packages/pyModeS/__init__.py", line 11, in <module>
    from .decoder import tell
  File "/usr/local/lib/python2.7/dist-packages/pyModeS/decoder/__init__.py", line 5
    print("%20s: " % label, end="")
                               ^
SyntaxError: invalid syntax

**Comment by junzis** [comment, marked as answer]
Hi @esnelling, the issue is indeed the Python version. `pyModeS` only works with Python 3 now.

I think you can use `update-alternatives` in your Linux console to change the default Python version to 3. Then `modeslive` should work.

**Reply by esnelling** [reply]
@junzis Thank you sir.  I have done that and modeslive works just fine now!!! Thanks so much.  You have done a great job in documentation.

**Comment by myvilli** [comment]
Hello,

I am trying to install pyModeS, and I am running in the same kind of issue :

~$ pip3 install pymodes
Collecting pymodes
  Using cached pymodes-2.14.tar.gz (392 kB)
ERROR: Exception:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pip/_internal/cli/base_command.py", line 186, in _main
    status = self.run(options, args)
  File "/usr/lib/python3/dist-packages/pip/_internal/commands/install.py", line 357, in run
    resolver.resolve(requirement_set)

~$ pip3 -V
pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)

I cannot install this package using pip3 (python3) , I have tried cleaning cache directory, also, installing with no-cache option

I want to run it on Ubuntu computer.

Any help would be appreciated.

**Reply by xoolive** [reply]
Can you try updating your pip version? It's working on my side with version 22 (but it should work with 21 as well)

If not, the conda version should appear very soon

**Reply by xoolive** [reply]
Indeed @myvilli, pip supports this format since 21.3

You may read more about it here: https://stackoverflow.com/questions/62983756/what-is-pyproject-toml-file-for

**Reply by myvilli** [reply]
@xoolive
You were right, the pip version installed was outdated.
Solved by updating pip with :
$ pip3 install --upgrade pip

Thank you.
